pkg/apimachinery/apis/common/v0alpha1/resource.go

```diff
@@ -26,10 +26,13 @@ func NewResourceInfo(group, version, resourceName, singularName, kind string,
 	return ResourceInfo{group, version, resourceName, singularName, shortName, kind, newObj, newList}
 }
 
-func (info *ResourceInfo) WithGroupAndShortName(group string, shortName string) ResourceInfo {
+func (info *ResourceInfo) WithGroupAndShortName(group, version, shortName string) ResourceInfo {
+	if version == "" {
+		version = info.version
+	}
 	return ResourceInfo{
 		group:        group,
-		version:      info.version,
+		version:      version,
```

Comment:
I am not sure I understand why this change

Reply:
used here:  https://github.com/grafana/grafana/pull/88041/files#diff-4f8682b7549ec2d252d819509508d94153e4e75210d627b625ff95800af6f2dfR169

it sets the version rather than locked at v0alpha1